test/models/user_test.rb

```diff
@@ -87,4 +87,13 @@ class UserTest < ActiveSupport::TestCase
       users(:keith).update!(last_name: nil)
     end
   end
+
+  test "it treats blank String API keys as nil to allow for default keys" do
+    stub_features(default_llm_keys: false) do
+      user = users(:keith)
+      user.update(openai_key: '', anthropic_key: ' ')
```

Comment:
@matthewbennink after merging in I noticed this is missing a bang. The update can fail and then this test mistakenly passes. Years ago I automatized almost always doing using bang on update and save since we typically expect it to succeed and want it to fail loudly if it doesn't (and controllers handle exceptions gracefully).

I'll quickly fix this in main, so just an FYI for you